adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -437,11 +437,9 @@ public void acquireToken(Activity activity, String resource, String clientId,
     }
 
     /**
-     * acquireToken will start interactive flow if needed. It checks the cache
-     * to return existing result if not expired. It tries to use refresh token
-     * if available. If it fails to get token with refresh token, behavior will
-     * depend on options. If promptbehavior is AUTO, it will remove this refresh
-     * token from cache and fall back on the UI if activitycontext is not null.
+     * acquireToken will start an interactive auth flow to acquire new tokens 
+     * with the requested claims. If promptbehavior is AUTO, it will remove this refresh
```

Comment:
This actually seems _less_ clear to me - if AT is called we:

1. [Call acquireToken](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/477ff358e1cc40587a8a71344650990288defad9/adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java#L477)
2. Which calls [performAcquireTokenRequest](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/477ff358e1cc40587a8a71344650990288defad9/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L111)
3. Which calls [tryAcquireTokenSilent](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/477ff358e1cc40587a8a71344650990288defad9/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L333)
4. Which calls [shouldTrySilentFlow](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/477ff358e1cc40587a8a71344650990288defad9/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L352)
5. Which returns `true` [if no claims challenge is present `&& prompt == AUTO`](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/477ff358e1cc40587a8a71344650990288defad9/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L388)

What's going on here?

Without digging around in the code exhaustively, I don't remember anything about this removing RT's (unless they've expired/been revoked and cannot resolve the silent call) -- won't we need `FORCE` for this behavior? /cc @addev-ashish

Reply:
ADAL won't remove the tokens unless getting "invalid_grant" from the server.

Reply:
The old javadoc does not see correct to me. If AT/RT exists and we still need the UI prompt. we should call AT with prompt behavior with `FORCE_PROMPT`.

Reply:
If it is a claim challenge call, we do not need `FORCE_PROMPT` behavior. The AT call will skip the silent call and jump into the acquireTokenInteractiveFlow.

Reply:
Thanks @iambmelt @heidijinxujia, I'll update the text. I had left the auto part from the first note, but it seems like that piece was also misleading.

Reply:
Made a few changes. The promptbehavior option in this specific overload does have an unclear use. We should clarify and add it to the description of prompt parameter before merging this PR.